### PR TITLE
Update concepts->block  to reflect product updates

### PR DESCRIPTION
**Pull request by discdiver**
Some cleanup to reflect changes in the product.
Also makes the page more concise and consistent.

Examples of JSON and custom AwsCredentials could probably use an update before long, but that's beyond the scope of this PR.

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
	- If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [ ] If this pull request adds new functionality, it includes unit tests that cover the changes
- [X] This pull request includes a label categorizing the change e.g. `maintenance`, `fix`, `feature`, `enhancement`, `docs`.
  

For documentation changes:

- [ ] This pull request includes redirect settings in `netlify.toml` for files that are removed or renamed.

For new functions or classes in the Python SDK:

- [ ] This pull request includes helpful docstrings.
- [ ] If a new Python file was added, this pull request contains a stub page in the Python SDK docs and an entry in `mkdocs.yml` navigation.

### CODE REVIEW

**Comment by discdiver** [comment]
TODO: update screenshot.

File: docs/concepts/blocks.md [hunk 1]
```diff
@@ -36,136 +38,131 @@ Prefect Cloud and the Prefect server UI display a library of block types availab
...
+| [Slack Webhook](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.SlackWebhook)                    | `slack-webhook`                                          | Enables sending notifications via a provided Slack webhook.                      |
+| [SMB](/concepts/filesystems/#smb)                                                                                    | `smb`                                                    | Store data as a file on a SMB share.                                                     |
+| [String](/api-ref/prefect/blocks/system/#prefect.blocks.system.String)                                               | `string`                                                 | A block that represents a string.                                         |
+| [Twilio SMS](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.TwilioSMS)                          | `twilio-sms`                                             | Enables sending notifications via Twilio SMS.                                                       |
```
Comment (billpalombi) [inline docs/concepts/blocks.md:71]:
```suggestion
| [Twilio SMS](/api-ref/prefect/blocks/notifications/#prefect.blocks.notifications.TwilioSMS)                          | `twilio-sms`                                             | Sends notifications via Twilio SMS.                                                       |
```

File: docs/concepts/blocks.md [hunk 2]
```diff
@@ -5,29 +5,31 @@ tags:
...
-You may configure blocks through code or via the Prefect Cloud and the Prefect server UI.
+Block types are Python classes with a handy UI webform for configuration.
+Blocks are instantiation of these classes with specific values.
 
-You can access blocks for both configuring flow [deployments](/concepts/deployments/) and directly from within your flow code.
+Configure blocks through Python code or via a form in the UI.
+Access blocks for use in Python code.
 
-Prefect provides some built-in block types that you can use right out of the box. Additional blocks are available through [Prefect Integrations](/integrations/). To use these blocks you can `pip install` the package, then register the blocks you want to use with Prefect Cloud or a Prefect server.
+Blocks live on in Prefect Cloud or your self-hosted Prefect server instance.
```
[the remark this answers is not in the document]
Comment (discdiver) [reply docs/concepts/blocks.md:27]:
```suggestion
Block values are stored in Prefect Cloud or your self-hosted Prefect server instance.
```

**Comment by discdiver** [comment]
> First pass

Good stuff.

File: docs/concepts/blocks.md [hunk 3]
```diff
@@ -36,136 +38,131 @@ Prefect Cloud and the Prefect server UI display a library of block types availab
 
 ## Prefect built-in blocks
 
-Prefect provides a broad range of commonly used, built-in block types. These block types are available in Prefect Cloud and the Prefect server UI.
+Prefect provides a broad range of commonly used, built-in block types.
```
Comment (billpalombi) [inline docs/concepts/blocks.md:41]:
```suggestion
Commonly used block types come built-in to Prefect. 
```

File: docs/concepts/blocks.md [hunk 4]
```diff
@@ -36,136 +38,131 @@ Prefect Cloud and the Prefect server UI display a library of block types availab
...
+These block types are pre-installed in Prefect Cloud and a self-hosted Prefect server instance.
```
Comment (billpalombi) [inline docs/concepts/blocks.md:42]:
```suggestion
```

File: docs/concepts/blocks.md [hunk 5]
```diff
@@ -191,18 +189,19 @@ print(json_block.value["the-answer"]) #42
 ```
 
 !!! tip "Sharing Blocks"
```
Comment (billpalombi) [inline docs/concepts/blocks.md:191]: I think we can get rid of this tip.

File: docs/concepts/blocks.md [hunk 6]
```diff
@@ -342,7 +346,10 @@ Available metadata fields include:
 
 ### Nested blocks
 
-Block are composable. This means that you can create a block that uses functionality from another block by declaring it as an attribute on the block that you're creating. It also means that configuration can be changed for each block independently, which allows configuration that may change on different time frames to be easily managed and configuration can be shared across multiple use cases.
+Blocks are composable, meaning that blocks can be used within other blocks.
```
Comment (billpalombi) [inline docs/concepts/blocks.md:349]:
```suggestion
Blocks are composable - a block can be used within other blocks.
```

File: docs/concepts/blocks.md (hunk 3 above)
Comment (discdiver) [reply docs/concepts/blocks.md:41]:
```suggestion
Commonly used block types come built-in with Prefect. 
```

Comment (billpalombi) [reply docs/concepts/blocks.md]: Yeah that's better

File: docs/concepts/blocks.md [hunk 7]
```diff
@@ -191,18 +188,19 @@ print(json_block.value["the-answer"]) #42
 ```
 
 !!! tip "Sharing Blocks"
-    Blocks can also be loaded by fellow Workspace Collaborators, available on [Prefect Cloud](/ui/cloud/).
+    If using [Prefect Cloud](/cloud/), blocks can be loaded by workspace collaborators.
```
Comment (discdiver) [inline docs/concepts/blocks.md:191]:
```suggestion
```

File: docs/concepts/blocks.md [hunk 8]
```diff
@@ -191,18 +188,19 @@ print(json_block.value["the-answer"]) #42
 ```
 
 !!! tip "Sharing Blocks"
```
Comment (discdiver) [inline docs/concepts/blocks.md:190]:
```suggestion
```

**Review by billpalombi — APPROVED** [review]
lgtm

**Pull request merged**